modules/auxiliary/admin/http/ulterius_file_download.rb

```diff
@@ -52,6 +52,30 @@ def process_data(index, parse_data)
   end
 
   def inflate_parse(data)
+    # TODO: Confirm expected behavior
+=begin
+    modules/auxiliary/admin/http/ulterius_file_download.rb:55:40: W: Lint/AmbiguousAssignment: Suspicious assignment detected. Did you mean -=?
+        zi = Zlib::Inflate.new(window_bits =-15)
+                                           ^^
+    Zlib definition:
+
+      While this example will work, for best optimization review the flags for
+      your specific time, memory usage and output space requirements.
+      def initialize(level=DEFAULT_COMPRESSION, window_bits=MAX_WBITS, mem_level=DEF_MEM_LEVEL, strategy=DEFAULT_STRATEGY)
+        #This is a stub, used for indexing
+      end
+
+
+    Example:
+    def zlib(level=1, window_bits=2, mem_level=3, strategy=4)
+        puts "level #{level}"
+        puts "window_bits #{window_bits}"
+        puts "mem_level #{mem_level}"
+        puts "strategy #{strategy}"
+    end
+
+    zlib(window_bits =-15)
+=end
     zi = Zlib::Inflate.new(window_bits =-15)
```

Comment:
Will await a second pair of eyes on this, this is currently:
```
zi = Zlib::Inflate.new(-15)
```

Which is probably unintended, but it worked at the time of module landing I imagine.

Reply:
> 
> 
> Will await a second pair of eyes on this, this is currently:
> 
> ```
> zi = Zlib::Inflate.new(-15)
> ```
> 
> Which is probably unintended, but it worked at the time of module landing I imagine.

This doesn't look right.

I have no idea what the intention  of the original code was either.

```
$ cat test.rb 
#!/usr/bin/env ruby

def test(a = 1, b = 2, c = 3, d = 4)
  puts "a: #{a}"
  puts "b: #{b}"
  puts "c: #{c}"
  puts "d: #{d}"
end


test(b = 1234)
$ ./test.rb 
a: 1234
b: 2
c: 3
d: 4
```

Reply:
Looks like I accidentally grabbed the method definition for the `Deflate` method, the inflate one is much easier to work out:

```
        def initialize(window_bits = Zlib::MAX_WBITS)
            # ...
        end
```

Fixed now :+1: